tests/CppUTest/TestUTestMacro.cpp

```diff
@@ -226,30 +226,6 @@ IGNORE_TEST(UnitTestMacros, UNSIGNED_LONGLONGS_EQUAL_TEXTWorksInAnIgnoredTest)
     UNSIGNED_LONGLONGS_EQUAL_TEXT(1, 0, "Failed because it failed"); // LCOV_EXCL_LINE
 } // LCOV_EXCL_LINE
 
-#else
-
-static void LONGLONGS_EQUALFailsWithUnsupportedFeatureTestMethod_()
-{
-    LONGLONGS_EQUAL(1, 1);
-} // LCOV_EXCL_LINE
-
-static void UNSIGNED_LONGLONGS_EQUALFailsWithUnsupportedFeatureTestMethod_()
-{
-    UNSIGNED_LONGLONGS_EQUAL(1, 1);
-} // LCOV_EXCL_LINE
-
-TEST(UnitTestMacros, LONGLONGS_EQUALFailsWithUnsupportedFeature)
-{
-    fixture.runTestWithMethod(LONGLONGS_EQUALFailsWithUnsupportedFeatureTestMethod_);
-    CHECK_TEST_FAILS_PROPER_WITH_TEXT("\"CPPUTEST_USE_LONG_LONG\" is not supported");
-}
-
-TEST(UnitTestMacros, UNSIGNED_LONGLONGS_EQUALFailsWithUnsupportedFeature)
-{
-    fixture.runTestWithMethod(UNSIGNED_LONGLONGS_EQUALFailsWithUnsupportedFeatureTestMethod_);
-    CHECK_TEST_FAILS_PROPER_WITH_TEXT("\"CPPUTEST_USE_LONG_LONG\" is not supported");
-}
-
```

Comment:
These can also stay when the earlier #define it removed.

I do have a preference to that as it keeps the interface the same.

Reply:
They cannot.That's the whole problem this MR is trying to solve. These are the constraints we're trying to place on the `cpputest_[u]longlong` types:
 1. same size as native `long long` types to avoid ABI breakages.
 2. C linkage so that C++ and C code can interoperate.
 3. assignable from an integral constant so `[UNSIGNED_]LONGLONGS_EQUAL` can be called without native `long long` support.

Based on everything I've tried, we have to pick two; we can't accomplish all three. I chose to drop support for 3, since these calls fail at runtime anyway.

Reply:
Yes, it could be solved by declaring a cpputest_longlong in the test since it is anyways inside a #if and it doesn't need to be given a value for the test. It would then still properly test that it would give a failure. And I guess it might be worth it...

That said, in practice, because of (3), the same code will never work that works with longlong support will not compile for not longlong support. So, it is probably not that useful to keep it.

I'll merge it as is. If you feel it still useful to add these tests to test the proper failure, then they can be returned in a separate PR.